src/core/ext/transport/chttp2/transport/writing.c

```diff
@@ -206,9 +203,20 @@ bool grpc_chttp2_begin_write(grpc_exec_ctx *exec_ctx,
     }
   }
 
+  bool partial_write = false;
+
   /* for each grpc_chttp2_stream that's become writable, frame it's data
      (according to available window sizes) and add to the output buffer */
-  while (grpc_chttp2_list_pop_writable_stream(t, &s)) {
+  while (true) {
+    if (t->outbuf.length > 1024 * 1024) {
```

Comment:
Since this will eventually be a tuned parameter, how about at least making it a #define or something now so that it's identifiable in the future?